### ISSUE TITLE
Dealing with cross-version 'wobbly' order of implicit variables

### ISSUE THREAD

**Issue by JacquesCarette** [agda/agda#4313]
This is coming up in agda-categories where we make use of `variables`, which are extremely convenient.  But in a few cases, the order between 2.6.0.1 and 2.6.1 has changed. The problem is that this has changed in a file (Categories.Category.Monoidal.Closed.IsClosed) that takes ages to typecheck [and lots of memory and it looks like 1TB of virtual memory!], so debugging this is very painful.

The real 'bug' here is that one cannot specify implicits, by name, out of order.  So only one of `{X = X} {Y= Y}` and `{Y= Y} {X = X}` will work, even though they really ought to mean the same thing! This means that if there is a change of order between versions, one's code just breaks. The only solution seems to be to not use the 'variables' feature for those functions. Is that really the case? @jespercockx

**Comment by JacquesCarette** [comment]
[Note that this is a 'regression' that perhaps should also be fixed before 2.6.1 is released -- #4212)

**Comment by jespercockx** [comment]
I think this is the same issue as https://github.com/agda/agda/issues/3280?

The reason why the arguments are now in a different order is probably because of the fix of https://github.com/agda/agda/issues/3779, since this changed the sorting algorithm for generalized variables.

**Comment by jespercockx** [comment]
As it's likely that the change in order of generalized variables is intentional, I'm hesitant to mark this as a regression. I strongly agree that you should be able to pass the implicit arguments in any order you want, but realistically speaking this will not be implemented in 2.6.1.

**Comment by JacquesCarette** [comment]
Yes, it's the same as that issue. You'll notice that I called this one a 'regression', unlike my previous issue that I explicitly labelled a regression...  As I do agree that this is unfortunate rather than a showstopper. I think I'll be able to work around it.

**Comment by gallais** [comment]
#3447 seems relevant too!
